ruby/ql/test/query-tests/variables/UninitializedLocal/UninitializedLocal.rb

```diff
@@ -0,0 +1,37 @@
+def m
+    puts "m"
+end
+
+def foo
+    m # calls m above
+    if false
+        m = 0
+        m # reads local variable m
+    else
+    end
+    m  #$ Alert
+    m2 # undefined local variable or method 'm2' for main (NameError)
+end
+
+def test_guards
+    if (a = 3 && a)  #$ SPURIOUS: Alert
```

Comment:
This actually parses as `a = (3 && a)`, so the alert is correct. If you change it to `(a = 3) && a` the alert disappears.

Reply:
Ah, that is right, the alert is correct.